apisix/plugins/error-log-logger.lua

```diff
@@ -0,0 +1,140 @@
+-- (the "License"); you may not use this file except in compliance with
+-- the License.  You may obtain a copy of the License at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+--
+
+local core = require("apisix.core")
+local plugin_name = "error-log-logger"
+local errlog = require "ngx.errlog"
+local ngx = ngx
+local tcp = ngx.socket.tcp
+local select = select
+local type = type
+local string = string
+
+local timer
+local schema = {
+    type = "object",
+    properties = {},
+    additionalProperties = false,
+}
+
+local log_level = {
+    STDERR =    ngx.STDERR,
+    EMERG  =    ngx.EMERG,
+    ALERT  =    ngx.ALERT,
+    CRIT   =    ngx.CRIT,
+    ERR    =    ngx.ERR,
+    ERROR  =    ngx.ERR,
+    WARN   =    ngx.WARN,
+    NOTICE =     ngx.NOTICE,
+    INFO   =    ngx.INFO,
+    DEBUG  =    ngx.DEBUG
+}
+
+local _M = {
+    version = 0.1,
+    priority = 1091,
+    name = plugin_name,
+    schema = schema,
+}
+
+
+function _M.check_schema(conf)
+    return core.schema.check(schema, conf)
+end
+
+
+local function try_attr(t, ...)
+    local count = select('#', ...)
+    for i = 1, count do
+        local attr = select(i, ...)
+        t = t[attr]
+        if type(t) ~= "table" then
+            return false
+        end
+    end
+
+    return true
+end
+
+
+local function report()
+    local local_conf = core.config.local_conf()
+    local host, port
+    local timeout = 3
+    local keepalive = 3
+    local level = "warn"
+    if try_attr(local_conf, "plugin_attr", plugin_name) then
+        local attr = local_conf.plugin_attr[plugin_name]
+        host = attr.host
+        port = attr.port
+        level = attr.loglevel or level
+        timeout = attr.timeout or timeout
+        keepalive = attr.keepalive or keepalive
+    end
+    level = log_level[string.upper(level)]
+
+    local status, err = errlog.set_filter_level(level)
+    if not status then
+        core.log.warn("failed to set filter level by ngx.errlog, the error is :", err)
+        return
+    end
+
+    local sock, soc_err = tcp()
+    if not sock then
+        core.log.warn("failed to init the socket " .. soc_err)
+        return
+    end
+    sock:settimeout(timeout*1000)
+    local ok, err = sock:connect(host, port)
+    if not ok then
+        core.log.warn("connect to the server failed for " .. err)
+        return
+    end
+    local logs = errlog.get_logs(10)
+    while ( logs and #logs>0 ) do
+        for i = 1, #logs, 3 do
+            if logs[i] <= level then --ommit the lower log producted at the initial
+                local bytes, err = sock:send(logs[i + 2])
```

Comment:
https://github.com/apache/apisix/blob/master/apisix/utils/batch-processor.lua

I think we can use this way to send the error log.